src/context/AuthContext.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+  AuthError,
+  AuthResponse,
+  Session,
+  User,
+  UserAttributes,
+  UserResponse,
+} from '@supabase/supabase-js';
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import supabaseAdmin from '../supabase/createAdminClient';
+import supabase from '../supabase/createClient';
+
+export interface AuthState {
+  session: Session | null;
+  user: User | null;
+  isLoading: boolean;
+  signIn: (newSession: Session | null) => void;
+  signUp: (
+    email: string,
+    password: string,
+    options: object,
+  ) => Promise<AuthResponse>;
+  signInWithEmail: (email: string, password: string) => Promise<AuthResponse>;
+  resendVerification: (email: string) => Promise<AuthResponse>;
+  resetPassword: (email: string) => Promise<
+    | {
+        data: object;
+        error: null;
+      }
+    | {
+        data: null;
+        error: AuthError;
+      }
+  >;
+  updateUser: (attributes: UserAttributes) => Promise<UserResponse>;
+  signOut: () => Promise<void>;
+  deleteCurrentUser: (userId: string) => Promise<void>;
+}
+
+const AuthContext = createContext({} as AuthState);
+
+export function useSession() {
+  const value = useContext(AuthContext);
+
+  return value;
+}
+
+export function AuthContextProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [session, setSession] = useState<Session | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    supabase.auth
+      .getSession()
+      .then(({ data: { session: newSession } }) => {
+        setSession(newSession);
+        setUser(newSession ? newSession.user : null);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+
+    supabase.auth.onAuthStateChange((_event, newSession) => {
+      setSession(newSession);
+    });
+  }, []);
+
+  const signIn = (newSession: Session | null) => {
+    setSession(newSession);
+  };
+
+  const signInWithEmail = async (email: string, password: string) => {
```

Comment:
If we're moving these functions to the auth context, should we delete the duplicate functionality from the `queries/auth.ts` file?